srt-protocol/src/protocol/sender/mod.rs

```diff
@@ -409,26 +378,15 @@ impl Sender {
         // TODO: reset EXP
     }
 
-    fn handle_handshake_packet(&mut self, handshake: HandshakeControlInfo, now: Instant) {
-        if let Some(control_type) = self.handshake.handle_handshake(&handshake) {
+    pub fn handle_handshake_packet(&mut self, handshake: HandshakeControlInfo, now: Instant) {
+        if let Some(control_type) = self.handshake.handle_handshake(handshake) {
```

Comment:
Agreed, I was incrementally refactoring, just hadn't gotten to it yet. I also intend to move all the connection Drain/Shutdown/Close logic to DuplexConnection as well, the keepalive timer too, and a few other things.

Reply:
Ok, now I want to leave this on the Sender for now because that DuplexConnection doesn't have the means for buffering packets to send yet, and maybe it never will. For this PR I'm going to just leave this the way it is. Eventually I will probably explore managing buffers externally using some form of context or dependency injection, perhaps via a "send" Fn, arena allocator, etc.